admin-dev/themes/new-theme/js/app/utils/serp/serp.vue

```diff
@@ -0,0 +1,140 @@
+
+<!--**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ *-->
+<template>
+  <div id="serp"">
+    <div class="serp-preview">
+      <div class="serp-title">{{title}}</div>
+      <div class="serp-url">{{url}}<span class="serp-arrow"></span></div>
+      <div class="serp-description">{{description}}</div>
+    </div>
+  </div>
+</template>
+
+<script>
+  export default {
+    name: 'serp',
+    data: () => ({
+        title: '',
+        url: '',
+        description: '',
+        defaultUrl: 'http://example.com/',
+    }),
+    methods: {
+        setTitle(title) {
+            if(title.length > 70) {
+                title = title.substring(0, 70) + ' ...';
+            }
+            this.title = title;
+        },
+        setUrl(url) {
+            this.url = url || this.defaultUrl;
+        },
+        setDescription(description) {
+            if(description.length > 150) {
+                description = description.substring(0, 150) + ' ...';
+            }
+            this.description = description;
+        }
+    }
+  };
+</script>
+
+<style lang="sass" type="text/scss">
```

Comment:
Not sure it's a good or bad idea, but you can import a file instead:

```js
<style lang="scss">
@import 'where/you/want/serp.css';
</style>
```

Reply:
VueJs style guide https://vuejs.org/v2/style-guide/#Component-style-scoping-essential

Reply:
and?

Reply:
it's just easier to play with component and keep specific style in its context ^^

Reply:
maybe, I don't know and it's certainly not me who will try to customize the design of our back office :)